assets/src/stories-editor/blocks/amp-story-cta/edit.js

```diff
@@ -8,128 +8,97 @@ import PropTypes from 'prop-types';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
-import {
-	Dashicon,
-	IconButton,
-} from '@wordpress/components';
-import {
-	URLInput,
-	RichText,
-} from '@wordpress/block-editor';
+import { useState, useEffect } from '@wordpress/element';
+import { Dashicon, IconButton } from '@wordpress/components';
+import { URLInput, RichText } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import './edit.css';
-import { select } from '@wordpress/data';
 import { getUniqueId, setInputSelectionToEnd } from '../../helpers';
 import { getBackgroundColorWithOpacity } from '../../../common/helpers';
 import { DraggableText } from '../../components';
 
-class CallToActionEdit extends Component {
-	constructor( props ) {
-		super( props );
-
-		if ( ! props.attributes.anchor ) {
-			this.props.setAttributes( { anchor: getUniqueId() } );
-		}
-
-		this.state = {
-			isEditing: false,
-			hasOverlay: true,
-		};
-
-		this.nodeRef = null;
-	}
-
-	bindRef = ( node ) => {
-		if ( ! node ) {
-			return;
-		}
-		this.nodeRef = node;
-	}
-
-	toggleIsEditing = ( enable ) => {
-		if ( enable !== this.state.isEditing ) {
-			this.setState( {
-				isEditing: ! this.state.isEditing,
-			} );
-		}
-	}
-
-	toggleOverlay = ( add ) => {
-		if ( add !== this.state.hasOverlay ) {
-			this.setState( {
-				hasOverlay: ! this.state.hasOverlay,
-			} );
+const CallToActionEdit = ( {
+	attributes,
+	backgroundColor,
+	className,
+	clientId,
+	fontSize,
+	isSelected,
+	name,
+	setAttributes,
+	textColor,
+} ) => {
+	const {
+		anchor,
+		text,
+		url,
+		customBackgroundColor,
+		opacity,
+		btnPositionTop,
+		btnPositionLeft,
+	} = attributes;
+
+	const [ isEditing, setIsEditing ] = useState( false );
+	const [ hasOverlay, setHasOverlay ] = useState( true );
+
+	useEffect( () => {
+		if ( ! anchor ) {
+			setAttributes( { anchor: getUniqueId() } );
 		}
-	}
+	}, [ anchor, setAttributes ] );
```

Comment:
Hmm, why is `setAttributes` tracked here? Isn't `setAttributes` always the same?

Reply:
The ESLint rule for React hooks will complain if it's missing. See https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies.

We can't really guarantee that `setAttributes` is always the same as it's outside our control, but adding it as a dependency here doesn't hurt either way.

Reply:
Okay, good to know that there is a ESLint rule that complains about it! 

It looked odd to me to add the function as well especially since `anchor` is already being tracked.
In this case most of the other comments (if not all) I had can be omitted as well!